static/src/js/components/KeyboardShortcutsModal/KeyboardShortcutsModal.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { Container, Row, Col } from 'react-bootstrap'
+import EDSCModal from '../EDSCModal/EDSCModal'
+
+export class KeyboardShortcutsModal extends Component {
+  constructor(props) {
+    super(props)
+    this.keyboardShortcuts = {
+      toggleModal: '?',
+      escapeModal: 'Escape'
+    }
+    this.keyboardShortcutRef = React.createRef()
+    this.onWindowKeyDown = this.onWindowKeyDown.bind(this)
+  }
+
+  componentDidMount() {
+    window.addEventListener('keydown', this.onWindowKeyDown)
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('keydown', this.onWindowKeyDown)
+  }
+
+  onWindowKeyDown(e) {
+    const { key } = e
+    const { keyboardShortcuts } = this
+    const { isOpen, onToggleKeyboardShortcutsModal } = this.props
+    // Toggle the modal when question key pressed
+    if (key === keyboardShortcuts.toggleModal) {
+      onToggleKeyboardShortcutsModal(!isOpen)
+
+      e.preventDefault()
+      e.stopPropagation()
+    } else if (key === keyboardShortcuts.escapeModal) {
+      if (isOpen) onToggleKeyboardShortcutsModal(false)
+
+      e.preventDefault()
+      e.stopPropagation()
+    }
+  }
+
+  render() {
+    const { isOpen, onToggleKeyboardShortcutsModal } = this.props
+    const keyboardShortcutsList = {
+      ']': 'Toggle main search result/granules panel',
+      g: 'Toggle granule filters panel',
+      a: 'Toggle advanced search modal',
+      '/': 'Focus on search input field',
+      '?': 'Display all keyboard shortcuts',
+      t: 'Toggle timeline'
+    }
+    return (
+      <EDSCModal
```

Comment:
Apologies if I mislead you a bit, if you grep through the codebase a bit you'll see that we have a container that we use `EDSCModalContainer` that will prevent you from needing to send some of these additional, empty, props. I suspect you'll end up with something along the lines of

```
      <EDSCModalContainer
        className="modal__keyboardShortcut_modal"
        isOpen={isOpen}
        size="md"
        id="keyboard-shortcuts"
        onModalHide={() => onToggleKeyboardShortcutsModal(false)}
        body={(
          <Container>
            {Object.entries(keyboardShortcutsList)
              .map(arr => (
                <Row key={arr[0]} className="mb-1">
                  <Col xs={1}>
                    <kbd>{arr[0]}</kbd>
                  </Col>
                  {':'}
                  <Col>{arr[1]}</Col>
                </Row>
              ))}
          </Container>
        )}
        title="Keyboard Shortcuts"
      />
```

Reply:
No issues, partly it was my fault also should have figured it out before :laughing: